hathor/consensus/poa/poa.py

```diff
@@ -46,3 +46,15 @@ def calculate_weight(settings: PoaSettings, block: PoaBlock, signer_index: int)
     """Return the weight for the given block and signer."""
     is_in_turn_flag = is_in_turn(settings=settings, height=block.get_height(), signer_index=signer_index)
     return BLOCK_WEIGHT_IN_TURN if is_in_turn_flag else BLOCK_WEIGHT_OUT_OF_TURN
+
+
+def get_signer_index_and_public_key(settings: PoaSettings, signer_id: bytes) -> tuple[int, bytes] | None:
```

Comment:
Maybe return a list with all matches? Even though it is unlikely to have multiple matches, it is a possibility. Just returning the first match might be misleading or cause bugs.

Reply:
This function has been removed. See https://github.com/HathorNetwork/hathor-core/pull/1071#discussion_r1669469069

Reply:
Ok. Thanks!